dojo/models.py

```diff
@@ -2110,8 +2122,9 @@ def save(self, dedupe_option=True, false_history=False, rules_option=True,
             from dojo.utils import do_apply_rules
             do_apply_rules(self, *args, **kwargs)
 
-        from dojo.utils import calculate_grade
-        calculate_grade(self.test.engagement.product)
+        if product_grading_option:
+            from dojo.utils import calculate_grade
```

Comment:
could we keep the imports at the start of the file?

Reply:
no due to circular dependencies

Reply:
circular dependencies could shows a bad code organization/splitting pb but ok.